packages/hebao_v1/contracts/base/MetaTxModule.sol

```diff
@@ -421,41 +435,39 @@ abstract contract MetaTxModule is BaseModule
 
     // ---- private functions -----
 
-    function getSigners(
-        address wallet,
-        bytes   memory data
+    function checkSigners(
+        address   wallet,
+        bytes     memory data,
+        address[] memory signers
         )
         private
         view
-        returns (address[] memory signers)
+        returns (bool)
     {
         bytes4 method = extractMethod(data);
         if (method == this.executeTransactions.selector) {
-            return extractAddressesFromCallData(data, 1);
+            address[] memory addresses = extractAddressesFromCallData(data, 1);
+            // Check if it matches the list of signers
+            if (addresses.length == signers.length) {
```

Comment:
should we simply make sure addresses[i] == signers[i]?

Reply:
So not check if the length of both lists are equal? If so, I'm not sure why not?

Reply:
Brecht, sorry this PR has been merged, if you feel we need to correct things, don't hesitate to create a new PR :)